### ISSUE TITLE
Correct small amounts of tile rotation

### ISSUE THREAD

**Issue by jil24** [sorgerlab/ashlar#127]
When doing cyclic IF, it's often hard to entirely eliminate all differences in rotation from cycle to cycle, whether it's coming from the camera or the stage holder. Often these are only 0.5-1 degree of rotation, but with large fields of view this can still cause stitched edges to appear blurred and probably causes worse problems when scanning larger areas. I can correct very small rotation errors manually, but it would be nice to have this as a part of the alignment.

**Comment by jmuhlich** [comment]
We have some experimental code for this but it's not quite ready for release. We assume that the camera and/or sample has rotated rigidly between cycles but there is no rotation _during_ imaging of a given cycle, so the rotation angle for all tiles will be the same. Does that match your requirements?

**Comment by jil24** [comment]
Yes, that’s exactly what is happening. There isn’t any significant rotation within a cycle.

**Comment by molnaresa** [comment]
Hi, is there maybe anything new regarding this issue? I have a very similar situation with 4i images, there is a 0.3-0.5 degree difference in rotation between some of the cycles.

The images are in CZI format, I’ve tried rotating manually using ZEN before running ASHLAR, but without success (ASHLAR throws an error when I use the rotated image as input: "ValueError: negative dimensions are not allowed"). @jil24 would you mind sharing how you corrected for your rotation errors that worked with ASHLAR afterwards?

**Comment by jil24** [comment]
My workflow leaves a lot to be desired, unfortunately. 

1. I extracted stage coordinates from my image metadata and converted everything to plain TIFs before starting.
2. I then used the pyStackReg package to compare cycle zero DAPI to each cycle’s DAPI for every single image tile. PyStackReg returns an affine transformation matrix for a rigid transformation which i converted to degrees of rotation, discarding the translation element. I then took the median between all tiles to get a single rotation value per-cycle and used it to rotate the image and crop everything down slightly to remove any border added during rotation. This was a small crop since the rotation was never more than 2-3 degrees. PyStackReg made a lot of registration errors, so it was important to take a median value to get rid of all the crazy outliers.
3. The hard part: generating gif files with stage position metadata in a format that ASHLAR would accept. I have some (bad) code I can share that might help with this.
